setters/transformers.go

```diff
@@ -0,0 +1,157 @@
+package setters
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+
+	"github.com/pkg/errors"
+)
+
+type (
+	intTransformer   func(int64, bool) (interface{}, error)
+	floatTransformer func(float64, bool) (interface{}, error)
+)
+
+func reflectIntTransformer(
+	v int64, ptr bool, kind reflect.Kind) (interface{}, error) {
+
+	fun := intFuncs(kind)
+	if ret, err := fun(v); err != nil {
+		return nil, err
+	} else {
+		if ptr {
+			return &ret, nil
+		}
+		return ret, nil
+	}
+}
+
+func reflectFloatTransformer(
+	v float64, ptr bool, kind reflect.Kind) (interface{}, error) {
+
+	fun := floatFuncs(kind)
+	if ret, err := fun(v); err != nil {
+		return nil, err
+	} else {
+		if ptr {
+			return &ret, nil
+		}
+		return ret, nil
+	}
+}
+
+func intTransformerFactory(t reflect.Kind) intTransformer {
+	return func(v int64, ptr bool) (interface{}, error) {
+		return reflectIntTransformer(v, ptr, t)
+	}
+}
+
+func floatTransformerFactory(t reflect.Kind) floatTransformer {
+	return func(v float64, ptr bool) (interface{}, error) {
+		return reflectFloatTransformer(v, ptr, t)
+	}
+}
+
+func intFuncs(kind reflect.Kind) func(int64) (interface{}, error) {
+	switch kind {
+	case reflect.Int:
+		return func(v int64) (interface{}, error) {
+			const (
+				MaxUint  = ^uint(0)
+				MaxRange = int64(int(MaxUint >> 1))
+				MinRange = int64(-int(MaxRange) - 1)
+			)
+			if v > MaxRange || v < MinRange {
+				return nil, fmt.Errorf(
```

Comment:
please do use "build on fly"errors it totally break the semantic.

you could create an error like:

```go
type ErrInvalidRange struct {
  k reflect.Kind
  v interface{}
}
```